public/apps/account/password-reset-panel.tsx

```diff
@@ -0,0 +1,157 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { useState } from 'react';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiIcon,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiTitle,
+} from '@elastic/eui';
+import { FormRow } from '../configuration/utils/form-row';
+import { AppDependencies } from '../types';
+import { getUserDetail, updateUser } from '../configuration/utils/internal-user-detail-utils';
+import { InternalUserUpdate } from '../configuration/types';
+import { buildUrl } from '../configuration/utils/url-builder';
+
+interface PasswordResetPanelProps extends AppDependencies {
+  username: string;
+  handleClose: () => void;
+}
+
+export function PasswordResetPanel(props: PasswordResetPanelProps) {
+  const [currentPassword, setCurrentPassword] = useState<string>('');
+  // reply on backend response of login call to verify
+  const [isCurrentPasswordInvalid, setIsCurrentPasswordInvalid] = useState<boolean>(false);
+
+  const [newPassword, setNewPassword] = useState<string>('');
+  // reply on backend response of user update call to verify
+  const [isNewPasswordInvalid, setIsNewPasswordInvalid] = useState<boolean>(false);
+
+  const [repeatNewPassword, setRepeatNewPassword] = useState<string>('');
+  const [isRepeatNewPasswordInvalid, setIsRepeatNewPasswordInvalid] = useState<boolean>(false);
+
+  const handleReset = async () => {
+    // validate the current password
+    try {
+      await props.coreStart.http.post('/auth/login', {
+        body: JSON.stringify({
+          username: props.username,
+          password: currentPassword,
+        }),
+      });
+    } catch (e) {
+      setIsCurrentPasswordInvalid(true);
+      return;
+    }
+
+    // update new password
+    try {
+      const user = await getUserDetail(props.coreStart.http, props.username);
+
+      const updateObject: InternalUserUpdate = {
+        password: newPassword,
+        backend_roles: user.backend_roles,
+        attributes: user.attributes,
+      };
+
+      await updateUser(props.coreStart.http, props.username, updateObject);
+
+      // need to log in with new password
+      window.location.href = buildUrl();
```

Comment:
I think need to log out. Otherwise user is probably still able to access if they know the url.

Reply:
by going back to home page, it will show a log in page again.

Reply:
Sure, I mean even goes back to login page, since the cookie is not cleared, I guess user access kibana via url.

Reply:
Tried accessing using some urls. It will redirect to the login page as well.

Reply:
I have a PR for logout functionality now.
https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/pull/368

I think buildUrl() is not safe for some cases. E.g. the url is `/_plugin/kibana/app/kibana/`, it will redirect to `/` if the server does not redirect it to `/_plugin/kibana/`, this doesn't work.

Reply:
Merged the aforementioned change and switched to logout